packages/modules/ad-personalization/src/worker/AdPersonalizationWorker.ts

```diff
@@ -34,18 +33,28 @@ export class AdPersonalizationWorker {
 
   async onPopupMessage(request: Message, sender: chrome.runtime.MessageSender, sendResponse: (response?: any) => void) {
     if (request.action === 'webmunkExt.popup.checkSettingsReq') {
-      const { response, tabId } = await this.send(request.data);
-      if (!response) return;
-      this.eventEmitter.emit(moduleEvents.AD_PERSONALIZATION, { data: request.data, value: response });
+      const url = await this.getAccordantUrl(request.key);
+      const { response, tabId } = await this.send(request.key, url);
+
+      if (response.error) {
```

Comment:
Please create separate error messages for the invalid URL case and for the case when the selector fails to find an element.